hpy/devel/src/runtime/ctx_type.c

```diff
@@ -310,6 +317,36 @@ create_slot_defs(HPyType_Spec *hpyspec)
     return result;
 }
 
+static PyBufferProcs*
+create_buffer_procs(HPyType_Spec *hpyspec)
+{
+    PyBufferProcs *buffer_procs = NULL;
+    if (hpyspec->defines != NULL) {
+        for (int i = 0; hpyspec->defines[i] != NULL; i++) {
+            HPyDef *src = hpyspec->defines[i];
+            if (src->kind != HPyDef_Kind_Slot)
+                continue;
+            switch (src->slot.slot) {
+                case HPy_bf_getbuffer:
+                    if (buffer_procs == NULL) {
+                        buffer_procs = PyMem_Calloc(1, sizeof(PyBufferProcs));
+                    }
+                    buffer_procs->bf_getbuffer = src->slot.cpy_trampoline;
+                    break;
+                case HPy_bf_releasebuffer:
+                    if (buffer_procs == NULL) {
+                        buffer_procs = PyMem_Calloc(1, sizeof(PyBufferProcs));
+                    }
+                    buffer_procs->bf_releasebuffer = src->slot.cpy_trampoline;
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+    return buffer_procs;
+}
```

Comment:
I seem to understand that this function is only needed on py<=3.8. It is worth a comment explaining why, and a note saying that we can kill it once we support only >=3.9

Reply:
Yes, this hack isn't needed on 3.9+. And I think it'd be better to already implement things correctly on 3.9, using a few `#if PY_VERSION_HEX < 0x03090000`, but that would require us testing on 3.9 first...